test_scripts/pypeit_test.py

```diff
@@ -0,0 +1,1159 @@
+
+        # Also try to find and fix pypeit files for
+        # spectroscopic standards.  This is a KLUDGE for
+        # gemini_gmos
+        setup.std_pyp_file = fix_pypeit_file_directory(setup.dev_path,
+                                                       raw_data_dir(),
+                                                       setup.instr,
+                                                       setup.name,
+                                                       setup.rdxdir,
+                                                       std=True)
+
+    # Only want to prep
+    if pargs.prep_only:
+        return setup
+
+    # ----------------------------------------------------------
+    # Reduce tests
+    # ----------------------------------------------------------
+    # Run pypeit
+    if not (flg_after or flg_ql):
+
+        setup.tests.append(PypeItReduceTest(setup, pargs.masters))
+
+        # Run pypeit on any standards, if they exist
+        if setup.std_pyp_file:
+            setup.tests.append(PypeItReduceTest(setup, pargs.masters, std=True))
+
+        # Try re-running keck_lris_red setups with masters
+        if setup.instr == 'keck_lris_red':
+            setup.tests.append(PypeItReduceTest(setup, True))
+
+    if pargs.tests == 'reduce':
+        # Skip the afterburner tests
+        return setup
+    # ----------------------------------------------------------
+
+    if not flg_ql:
+        # ----------------------------------------------------------
+        # SensFunc tests
+        # ----------------------------------------------------------
+        # UVIS algorithm without .sens file
+        if setup.instr == 'shane_kast_blue' and '600' in setup.name:
+            # TODO: Instead check for file with standard star in the
+            # file name?
+            std_file = 'spec1d_b24-Feige66_KASTb_2015May20T041246.960.fits'
+            setup.tests.append(PypeItSensFuncTest(setup, std_file=std_file))
+
+        # IR algorithm with .sens file
+        if setup.instr == 'gemini_gnirs' and setup.name == '32_SB_SXD':
+            std_file = 'spec1d_cN20170331S0206-HIP62745_GNIRS_2017Mar31T083351.681.fits'
+            sens_file = os.path.join(setup.dev_path, 'sensfunc_files', 'gemini_gnirs_32_sb_sxd.sens')
+            setup.tests.append(PypeItSensFuncTest(setup, std_file=std_file, sens_file=sens_file))
+
+        # IR algorithm, multi-slit, without .sens file
+        if setup.instr == 'gemini_gmos' and setup.name == 'GS_HAM_R400_860':
+            std_file = 'spec1d_S20181219S0316-GD71_GMOS-S_1864May27T230832.356.fits'
+            setup.tests.append(PypeItSensFuncTest(setup, std_file=std_file))
+
+        # ----------------------------------------------------------
+
+        # ----------------------------------------------------------
+        # fluxing/coadd1d/tellfit tests
+        # ----------------------------------------------------------
+        flux_coadd1d_pairs = {'shane_kast_blue': '600_4310_d55',  # long-slit
```

Comment:
I like this and wonder if it should be set at the top so we don't have
to dig into here to modify.